test/tests/classTransformer.test.ts

```diff
@@ -0,0 +1,118 @@
+      email: 'somebody@gmail.com',
+      password: 'secret',
+    } as Account)
+    // note here that _id is an ObjectId, hence the toString()
+    // otherwise it will have the shape of : { _bsonType: 'ObjectId', id: ArrayBuffer }
+    id = _id.toString()
+  })
+  describe('lean', () => {
+    it('should be able to use class-transformer', async () => {
+      // lean return a Plain Old Javascript Object
+      const POJO = await AccountModel.findById(id).lean()
+      // deserialize Plain Old Javascript Object into an instance of the Account class
+      const deserialized = plainToClass(Account, POJO)
+      // serialize Account instance back to a Plain Old Javascript Object, applying class-transformer's magic
+      const serialized = classToPlain(deserialized)
+      // the reason for doing a transformation round-trip here
+      // is that class-transformer can only works it magic on an instance of a class with its annotation
+      expect(serialized)
+      .toStrictEqual({
+        _id: id,
+        __v: 0,
+        email: 'somebody@gmail.com',
+      })
+    })
+    it(`should be able to use class-transformer's groups`, async () => {
+      // lean return a Plain Old Javascript Object
+      const POJO = await AccountModel.findById(id).lean()
+      const options = { groups: ['admin'] }
+      // deserialize Plain Old Javascript Object into an instance of the Account class
+      const deserialized = plainToClass(Account, POJO, options)
+      // serialize Account instance back to a Plain Old Javascript Object, applying class-transformer's magic
+      const serialized = classToPlain(deserialized, options)
+      // the reason for doing a transformation round-trip here
+      // is that class-transformer can only works it magic on an instance of a class with its annotation
+      expect(serialized)
+      .toStrictEqual({
+        _id: id,
+        __v: 0,
+        email: 'somebody@gmail.com',
+        password: 'secret',
+      })
+    })
+  })
+  describe('exec', () => {
+    it('should be able to use class-transformer', async () => {
+      // exec return a Mongoose Object
+      const MO = await AccountModel.findById(id).exec()
+      // deserialize Mongoose Object into an instance of the Account class
+      const deserialized = plainToClass(Account, MO)
+      // serialize Account instance back to a Plain Old Javascript Object, applying class-transformer's magic
+      const serialized = classToPlain(deserialized)
+      // the reason for doing a transformation round-trip here
+      // is that class-transformer can only works it magic on an instance of a class with its annotation
+      expect(serialized)
+      .toStrictEqual({
+        _id: id,
+        __v: 0,
+        email: 'somebody@gmail.com',
+      })
+    })
+    it(`should be able to use class-transformer's groups`, async () => {
+      // exec return a Mongoose Object
+      const MO = await AccountModel.findById(id).exec()
```

Comment:
```suggestion
      const MO = await AccountModel.findById(id).orFail().exec()
```

to remove the `| undefined` from the type and ensure the test fails if the id cannot be found